File: IEEExtreme/other_problems/rumour/rumour.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    unsigned long long q,a,b,ha,hb,maxh,minh,maxn,minn,c;
    scanf("%llu", &q);
    for (unsigned long long i = 0; i < q; i++){
        scanf("%llu%llu", &a, &b);
        ha = log2(a);
        hb = log2(b);
        if (ha < hb) {
    		maxh = hb;
    		maxn = b;
    		minh = ha;
    		minn = a;
	    } else {
		maxh = ha;
		maxn = a;
		minh = hb;
		minn = b;
	    }
		c = 0;
		while (minh<maxh){
		    maxn = maxn/2;
		    maxh--;
		    c++;
		}
		
		while (minn!=maxn){
		    maxn = maxn/2;
		    minn = minn/2;
		    c += 2;
		}
		printf("%llu\n",c);
    }
    

return 0;
}

File: IEEExtreme/other_problems/rumour/rumour.py
from math import log 

q = int(input())
for i in range(q):
	ab = input()
	ab = ab.split(" ")
	a = int(ab[0])
	b = int(ab[1])
	ha = int(log(a, 2))
	hb = int(log(b, 2))
	if ha < hb:
		maxh = hb
		maxn = b
		minh = ha
		minn = a
	else:
		maxh = ha
		maxn = a
		minh = hb
		minn = b

	c = 0
	
	for j in range(1, maxh-minh+1):
	    maxn = int(maxn/2)
	    c += 1 
	

	while minn != maxn:
		maxn = int(maxn/2)
		minn = int(minn/2)
		c += 2

	print(c)
